sdk/core/Azure.Core/src/Cryptography/IKeyEncryptionKeyResolver.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core.Cryptography
+{
+    /// <summary>
+    /// An object capable of retrieving key encryption keys from a provided key identifier.
+    /// </summary>
+    public interface IKeyEncryptionKeyResolver
+    {
+        /// <summary>
+        /// Retrieves the key encryption key corresponding to the specified keyId
+        /// </summary>
+        /// <param name="keyId">The key idenitifier of the key encryption key to retrieve</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>The key encryption key corresponding to the specified keyId</returns>
+        IKeyEncryptionKey Resolve(string keyId, CancellationToken cancellationToken);
```

Comment:
Nit - what happens when it's not resolved?  Do we return null or throw something?  It might be helpful to document that here.